### DISCUSSION TITLE
Consultative transfer using .net sdk and the current call status

### DISCUSSION THREAD

**Discussion by bfany365** [graphql-dotnet/graphql-dotnet#4084] [Q&A]
Hi.

I am using Graph API .net sdk to accept incoming call. Lets say the incoming call is A and then I do make another call to ask if an agent can join (call it B) to call A. 

This works fine and I can see the participant in call A are now in a call with participant in B.

```
var transferTarget1 = new InvitationParticipantInfo {
ReplacesCallId = call_B.Id.ToString(),
Identity = agentIdinCallB
};

await (_callA).TransferAsync(transferTarget1);
```

However, I see that _callA is Terminated now I tried the other way :

```
var transferTarget1 = new InvitationParticipantInfo {
ReplacesCallId = call_A.Id.ToString(),
Identity = agentIdinCallB
};
await (_callB).TransferAsync(transferTarget1);
```

Same happend and incoming call, A is terminated.
I need to keep track of the incoming call for Observability proposes and later add participants to it, but now it does not let me do that. Does calling TransferAsync makes new a call object ?

**Comment by Shane32** [comment]
I’m a little confused. Does this relate to the Microsoft Graph API?

**Reply by bfany365** [reply]
There was a typo in question. I corrected it. 

I am talking about this API :

https://learn.microsoft.com/en-us/graph/api/call-transfer?view=graph-rest-1.0&tabs=csharp

**Comment by Shane32** [comment]
This repository is for a NuGet package that allows developers to host their own GraphQL API. The Microsoft Graph API is a REST API and does not use GraphQL.
